portfolio/src/main/webapp/js/script.js

```diff
@@ -32,77 +20,167 @@ function myTimer() {
+  fetch("/list-comments")
+    .then(response => response.json())
+    .then((comments) => {
+      const commentContainer = document.getElementById('comment-section');
+      commentContainer.innerHTML = "";
+      comments.forEach((comment) => {
+        commentContainer.appendChild(createCommentItem(comment))
+      });
+    });
+}
 
 /**
- * Inject HTML template code.
+ * Create a text HTML element
  */
-function htmlInject(templatePath, htmlTarget) {
-    return fetch(templatePath)
-        .then(response => {
-            return response.text();
-        })
-        .then(text => {
-            document.querySelector(htmlTarget).innerHTML = text;
-        }) 
+function createHTML(elementType, content) {
+    const htmlElement = document.createElement(elementType);
+    htmlElement.innerHTML = content;
+    return htmlElement;
 }
 
-function onload() {
-    htmlInject('../header.html', ".meta-header")
-        .then(() => {
-            return htmlInject('../footer.html', ".footer")
-        })
-        .then(() => {
-            return setInterval(addRandomGreeting, 1000);
-        })
-        .then(() => {
-            return setInterval(myTimer, 1000);
-        })
-        .then(() => {
-            return setInterval(() => stumbleBuildString('MISSION', "#mission"), 50)
-        })
+/**
+ * Create a button
+ */
+function createButton(text, onclick) {
+    const buttonElement = document.createElement('button');
+    buttonElement.innerText = text;
+    buttonElement.addEventListener('click', onclick);
+    return buttonElement;
+}
+
+/**
+ * Render a single comment
+ */
+function createCommentItem(comment) {
+  const authorElement = createHTML('h5', comment.userName);
+  const likeElement = createHTML('h5', comment.like + " like");
+  const timeElement = createHTML('h5', comment.time);
+
+  let headerHTML = document.createElement('div');
+  headerHTML.className = "comment-row";
+  let headerElements = [authorElement, timeElement, likeElement]
+  headerElements.forEach((htmlElement) => {
+    headerHTML.appendChild(htmlElement)
+  });
+  
+  const contentElement = createHTML('h4', comment.content);
+
+  const deleteButtonElement = createButton('Delete', () => {
+    deleteComment(comment);
+    commentItem.remove();
+  });
+
+  const likeButtonElement = createButton('Like', () => likeComment(comment));
+
+  let footerHTML = document.createElement('div');
+  footerHTML.className = "comment-row";
+  let footerElements = [likeButtonElement, deleteButtonElement];
+  footerElements.forEach((htmlElement) => {
+    footerHTML.appendChild(htmlElement)
+  });
+
+  let commentHTML = document.createElement('div');
+  commentHTML.className = "comment";
+  
+  let commentElements = [headerHTML, contentElement, footerHTML];
+  commentElements.forEach((htmlElement) => {
+    commentHTML.appendChild(htmlElement)
+  });
+
+  return commentHTML;
 }
 
+/**
+ * Auxiliary string comparsion function
+ */
+function compareString(a, b) {
```

Comment:
I'd guess there is a js method you could use for that.

Reply:
you are right, turns out it was just `==`.  😂